shipyard-controller/handler/sequencedispatcher.go

```diff
@@ -80,7 +81,28 @@ func (sd *SequenceDispatcher) Add(queueItem models.QueueItem) error {
 		//if there are multiple shipyard we should only write
 		return sd.add(queueItem)
 	}
+}
 
+func (sd *SequenceDispatcher) addItemToQueue(queueItem models.QueueItem) error {
+	if err2 := sd.add(queueItem); err2 != nil {
+		return err2
+	}
+	return ErrSequenceBlockedWaiting
+}
+
+func (sd *SequenceDispatcher) isQueueEmpty(queueItem models.QueueItem) (bool, error) {
+	queuedSequences, err := sd.sequenceQueue.GetQueuedSequences()
```

Comment:
would we achieve the same functionality by checking if there are SequenceExecutionStates with status `triggered` (for the same project/stage/service)? `sequenceQueue.GetQueuedSequences()` will return the complete queue, and also this means that having an item form another project in the queue will affect if a sequence will be immediately dispatched or not

Reply:
sure we can try this approach

Reply:
applied